core/templates/pages/learner-dashboard-page/goals-tab.component.ts

```diff
@@ -82,13 +83,14 @@ export class GoalsTabComponent implements OnInit {
   editGoalsTopicClassification: number[] = [];
   editGoalsTopicBelongToLearntToPartiallyLearntTopic: boolean[] = [];
   windowIsNarrow: boolean = false;
-  showThreeDotsDropdown: boolean = false;
   directiveSubscriptions = new Subscription();
 
   ngOnInit(): void {
     this.MAX_CURRENT_GOALS_LENGTH = AppConstants.MAX_CURRENT_GOALS_COUNT;
     this.currentGoalsStoryIsShown = [];
+    this.showThreeDotsDropdown = [];
     this.currentGoalsStoryIsShown[0] = true;
+    this.showThreeDotsDropdown[0] = false;
```

Comment:
Could you please explain what these lines are intended to do? I'm a little unsure if these are right.

Reply:
if I just removes them then it shows a weird behaviour. shown below They are all correct as shown in proof everything is working fine.

[Screencast from 03-09-2023 10:21:54 AM.webm](https://user-images.githubusercontent.com/118007662/223921173-8184d99e-93bd-4275-8983-5b214abf6b88.webm)

I hope you understood what these lines do.

Reply:
> as shown in proof everything is working fine.

I'm sure it is. I just wish to know what exactly these lines accomplish (more specifically, what you intend for them to accomplish).

> I hope you understood what these lines do.

I don't, which is why I'm asking you :)

Reply:
Actually these lines ensures that , when we add a goal it should open with three dots and not with the dropdown of three dots and when I click on the three dots the function should work before introducing of these lines when I adds the goal it opens with the dropdown menu i.e. remove option and on clicking the three dots nothing happens , introducing these lines makes everything fine.

Reply:
Okay, so you're initializing the dropdown to stay closed upon first loading, but you need to do that for all the dropdowns, not just the first (index 0) one.

Set the length of the array to be equal to the total number of goals, then initalized every element of the array to be false.